app/src/main/java/com/example/habitsmasher/NotificationFragment.java

```diff
@@ -0,0 +1,255 @@
+                                        // accept the request
+                                        removeFollowRequestForUserInDatabase(_user.getId(),
+                                                followerID);
+                                        addNewFollowerForUserInDatabase(_user.getId(), followerID);
+                                        addUserToFollowingForUserInDatabase(followerID, _user.getId());
+                                        resetOptionsOfAdapter();
+                                        break;
+                                    case R.id.deny_request_button:
+                                        // deny the request
+                                        removeFollowRequestForUserInDatabase(_user.getId(),
+                                                _notificationItemAdapter._snapshots.get(position).getId());
+                                        resetOptionsOfAdapter();
+                                        break;
+                                }
+                            }
+                });
+        // connect listener to recycler view
+        recyclerView.addOnItemTouchListener(touchListener);
+    }
+
+    /**
+     * This readjusts the adapter of the RecyclerView so accepted/denied requests
+     * are deleted (as a consequence, when this occurs, any new requests
+     * sent after the user loads this fragment will also be loaded)
+     *
+     * Source:
+     * https://stackoverflow.com/questions/48014211/can-we-update-query-of-
+     * firestore-recycler-adapter-without-setting-the-whole-adap
+     */
+    protected void resetOptionsOfAdapter() {
+        _notificationItemAdapter.stopListening();
+        Query query = getListFromFirebase();
+        FirestoreRecyclerOptions<User> options = new FirestoreRecyclerOptions.Builder<User>()
+                                                                             .setQuery(query,
+                                                                                       User.class)
+                                                                             .build();
+        _notificationItemAdapter.updateOptions(options);
+        _notificationItemAdapter.startListening();
+    }
+
+    @Override
+    protected Query getListFromFirebase() {
+        DocumentReference sampleDocument = _db.collection("Users").document(_user.getId());
+        Task<DocumentSnapshot> querySnapshotTask = sampleDocument.get();
+
+        while (!querySnapshotTask.isComplete());
+
+        Map<String, Object> objectMap = querySnapshotTask.getResult().getData();
+
+        ArrayList<String> requestList = (ArrayList<String>) objectMap.get(FOLLOW_REQUEST_FIELD);
+        if(requestList != null && !requestList.isEmpty()) {
+            return _db.collection("Users").whereIn("id", requestList);
+        }
+
+        // return empty query when there are no follow requests
+        return _db.collection("Users").whereIn("id", Arrays.asList(""));
```

Comment:
This returns an empty query when a user has no follow requests (not null since setting it null caused some NPE problems)